### DISCUSSION TITLE
Automated accessibility tests

### DISCUSSION THREAD

**Discussion by benallpass** [MobileNativeFoundation/discussions#167] [General]
Hi everyone 👋 

I'm curious to hear what (if any) automated tests people are carrying out for catching accessibility issues.

I'm aware of the Espresso option mentioned in the developer docs [here](https://developer.android.com/guide/topics/ui/accessibility/testing#automated), and also Axe Android's [library](https://github.com/dequelabs/axe-android), but if anyone has any other tools/ideas/methods I'd be keen to hear!

Thanks a lot!

**Comment by dromerobarria** [comment]
Hi, Daniel here.

Im iOS developer from Chile, in the iOS platform we dont have a lot of tools to test accessibility ... right now we use Reveal App which  has an Accessibility Workspace, but other than that nothing else sadly. Will mention A11yUITests from 
Rob Whitaker as another way to test it, but it wont be maintain.

**Comment by mltbnz** [comment]
Hi. There is [AccessibilitySnapshot](https://github.com/cashapp/AccessibilitySnapshot) Currently in a POC phase for adding but looks very good

**Comment by parikhparth23** [comment]
Hello,
I work at a Social media company and we use a paid 3rd party tool - Evinced.
You can check Deque as welll.

With iOS 17 Apple has added support for a11y testing - https://developer.apple.com/videos/play/wwdc2023/10035/
